exec/csnex/Number.cs

```diff
@@ -23,9 +23,34 @@ public static Number FromString(string str)
             return new Number(Decimal.Parse(str, System.Globalization.NumberStyles.Float));
         }
 
+#region Static Arithmetic Functions
+        public static Number Subtract(Number x, Number y)
+        {
+            return new Number(x.val - y.val);
+        }
+#endregion
+
+        public bool IsInteger()
+        {
+            Decimal i = Decimal.Round(val, MidpointRounding.AwayFromZero);
+            return Decimal.Compare(val, i) == 0;
+        }
+
+        public bool IsNegative()
+        {
+            return Math.Sign(val) == -1;
+        }
+
+#region Overrides and Operators
+        public static implicit operator int(Number n)
```

Comment:
Do you really want this to be an implicit automatic conversion? It seems like an implicit conversion would be more likely to introduce accidental hidden errors because you can't obviously tell when it is happening (by looking at the code). Suggest making this an explicit method call, instead of an implicit operator. Also, a 32 bit result is insufficient.